dotcom-rendering/src/lib/useDeeplyReadTestVariant.tsx

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState } from 'react';
+import { deeplyReadRightColumn } from '../experiments/tests/deeply-read-right-column';
+import { useAB } from './useAB';
+
+type Variant =
+	| 'deeply-read-only'
+	| 'deeply-read-and-most-viewed'
+	| 'most-viewed-only'
+	| 'none';
+
+export const useDeeplyReadTestVariant = (): Variant => {
+	const [variant, setVariant] = useState<Variant>('none');
+	const ABTestAPI = useAB()?.api;
+
+	useEffect(() => {
+		const variantId =
+			(ABTestAPI?.runnableTest(deeplyReadRightColumn)?.variantToRun.id as
+				| Variant
```

Comment:
You might want to use [`isOneOf`](https://github.com/guardian/csnx/blob/022a2330dda928e3145e6f82bbc0f32c0b0d4182/libs/%40guardian/libs/src/isOneOf/README.md#L4) to avoid the type casting